dbt/adapters/sqlserver/connections.py

```diff
@@ -97,16 +126,55 @@ def open(cls, connection):
 
             con_str.append(f"Database={credentials.database}")
 
-            if not getattr(credentials, 'windows_login', False):
-                con_str.append(f"UID={credentials.UID}")
-                con_str.append(f"PWD={credentials.PWD}")
-            else:
+            type_auth = getattr(credentials, 'authentication', 'sql')
+
+            if 'ActiveDirectory' in type_auth:
+                con_str.append(f"Authentication={credentials.authentication}")
+
+                if type_auth == "ActiveDirectoryPassword":
+                    con_str.append(f"UID={{{credentials.UID}}}")
+                    con_str.append(f"PWD={{{credentials.PWD}}}")
+                elif type_auth == "ActiveDirectoryInteractive":
+                    con_str.append(f"UID={{{credentials.UID}}}")
+                elif type_auth == "ActiveDirectoryIntegrated":
+                    # why is this necessary???
+                    con_str.remove("UID={None}")
+                elif type_auth == "ActiveDirectoryMsi":
+                    raise ValueError("ActiveDirectoryMsi is not supported yet")
+
+            elif type_auth == 'ServicePrincipal':
+                app_id = getattr(credentials, 'AppId', None)
+                app_secret = getattr(credentials, 'AppSecret', None)
+
+            elif getattr(credentials, 'windows_login', False):
                 con_str.append(f"trusted_connection=yes")
+            elif type_auth == 'sql':
+                con_str.append("Authentication=SqlPassword")
+                con_str.append(f"UID={{{credentials.UID}}}")
+                con_str.append(f"PWD={{{credentials.PWD}}}")
+
+            if not getattr(credentials, 'encrypt', False):
+                con_str.append(f"Encrypt={credentials.encrypt}")
 
             con_str_concat = ';'.join(con_str)
```

Comment:
This code displays the credentials in the dbt logs without any encryption. So we would prefer this line be commented out and can be used only during debugging purposes.
This fix can be made in a separate PR but wanted to highlight you this scenario.